spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -133,6 +133,31 @@ public PubSubTemplate setMessageConverter(PubSubMessageConverter messageConverte
 		return publish(topic, this.messageConverter.toPayload(payload), headers);
 	}
 
+	@Override
+	public ListenableFuture<String> publish(String topic, String payload, Charset charset) {
+		return publish(topic, payload, null, charset);
+	}
+
+	@Override
+	public <T> ListenableFuture<String> publish(String topic, T payload) throws IOException {
+		ListenableFuture<String> result;
+
+		if (payload instanceof String) {
+			result = publish(topic, payload, null);
+		}
+		else if (payload instanceof byte[]) {
+			result = publish(topic, payload, null);
+		}
+		else if (payload instanceof ByteString) {
+			result = publish(topic, payload, null);
+		}
+		else {
+			result = publish(topic, payload, null);
```

Comment:
Why didn't you fix the ambiguity for the `publish()` with headers?

I mean why don't collapse them to a single generic one as well?